tokio/src/net/tcp/socket.rs

```diff
@@ -482,6 +482,48 @@ impl TcpSocket {
         let mio = self.inner.listen(backlog)?;
         TcpListener::new(mio)
     }
+
+    /// Converts a TcpStream into a TcpSocket
+    /// This converts the TcpStream into
+    /// the platform specific raw_fd(unix) / raw_socket(windows)
+    /// and then converts them into the TcpSocket
+    /// as TcpSocket implements the FromRawFd(unix) and
+    /// FromRawSocket(windows) traits.
+
+    /// # Examples
+    /// ```
+    /// use tokio::net::TcpSocket;
+    ///
+    /// use std::net::TcpStream;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> io::Result<()> {
+    ///
+    ///     let stream = TcpStream::connect("127.0.0.1:8080")?;
+    ///
+    ///     let socket = TcpSocket::from_std_stream(stream)?;
+    /// # drop(socket);
+    ///
+    ///     Ok(())
```

Comment:
I would like to use `socket2` in this example. Your example is currently converting a socket that is in a connected state, which you aren't supposed to do.